perfkitbenchmarker/benchmarks/cassandra_stress_benchmark.py

```diff
@@ -185,89 +340,118 @@ def _ResultFilePath(vm):
+                           population_params=None):
+  """Start all loader nodes as Cassandra clients and run stress test.
 
   Args:
-    benchmark_spec: The benchmark specification. Contains all data
-        that is required to run the benchmark.
+    cassandra_vms: list. A list of vm objects. Cassandra servers.
+    load_vms: list. A list of vm objects. Cassandra clients.
+    num_keys: integer. The number of operations cassandra-stress clients should
+        issue.
+    command: string. The cassandra-stress command to use.
+    profile_operations: string. The operations to use with user mode.
+    population_size: integer. The population size.
+    population_dist: string. The population distribution.
+    population_params: string. Representing additional population parameters.
   """
-  loader_vms = benchmark_spec.vm_groups[CLIENT_GROUP]
   num_loaders = len(loader_vms)
-  cassandra_vms = benchmark_spec.vm_groups[CASSANDRA_GROUP]
   data_node_ips = [vm.internal_ip for vm in cassandra_vms]
-  if not FLAGS.num_keys:
-    FLAGS.num_keys = NUM_KEYS_PER_CORE * cassandra_vms[0].num_cpus
-    logging.info(
-        'Num keys not set, using %s in cassandra-stress test.',
-        FLAGS.num_keys)
-  keys_per_vm = FLAGS.num_keys / num_loaders
-  if FLAGS.num_keys % num_loaders:
+  population_size = population_size or num_operations
+  operations_per_vm = int(math.ceil(float(num_operations) / num_loaders))
+  population_per_vm = population_size / num_loaders
+  if num_operations % num_loaders:
     logging.warn(
-        'Total number of keys rounded to %s (%s keys per loader vm).',
-        keys_per_vm * num_loaders, keys_per_vm)
+        'Total number of operations rounded to %s '
+        '(%s operations per loader vm).',
+        operations_per_vm * num_loaders, operations_per_vm)
   logging.info('Executing the benchmark.')
-  args = [((loader_vms[i], i, keys_per_vm, data_node_ips), {})
+  args = [((loader_vms[i], i, operations_per_vm, data_node_ips,
+            command, profile_operations, population_per_vm,
+            population_dist, population_params), {})
           for i in xrange(0, num_loaders)]
   vm_util.RunThreaded(RunTestOnLoader, args)
-
-
-def WaitForLoaderToFinish(vm):
-  """Watch loader node and wait for it to finish test.
-
-  Args:
-    vm: The target vm.
-  """
-  result_path = _ResultFilePath(vm)
-  while True:
-    resp, _ = vm.RemoteCommand('tail -n 1 ' + result_path)
-    if re.findall(r'END', resp):
-      break
-    if re.findall(r'FAILURE', resp):
-      vm.PullFile(vm_util.GetTempDir(), result_path)
-      raise errors.Benchmarks.RunError(
-          'cassandra-stress tool failed, check %s for details.'
-          % posixpath.join(vm_util.GetTempDir(),
-                           os.path.basename(result_path)))
-    time.sleep(SLEEP_BETWEEN_CHECK_IN_SECONDS)
+  logging.info('Tests running. Watching progress.')
+  vm_util.RunThreaded(WaitForLoaderToFinish, loader_vms)
```

Comment:
Are you still waiting for the loader to finish? The wait component should be handled by *RemoteCommand.